src/test_simple_graph.py

```diff
@@ -13,6 +13,9 @@ def test_add_node():
 def test_nodes():
     from simple_graph import SimpleGraph
     new_graph = SimpleGraph()
+    new_graph.add_edge('A', 'A', 10)
+    new_graph.add_edge('A', 'B', 20)
+    new_graph.add_edge('A', 'C', 30)
     new_graph.graph = {'A': ['A', 'B', 'C']}
     assert new_graph.nodes() == ['A']
```

Comment:
This still needs to produce all the nodes in the graph, not just the one. And there's no point in adding all of those edges if you're just going to re-set a new graph.